src/java_bytecode/java_types.h

```diff
@@ -44,6 +91,28 @@ inline java_class_typet &to_java_class_type(typet &type)
   return static_cast<java_class_typet &>(type);
 }
 
+class java_componentt : public struct_union_typet::componentt
+{
+public:
+  std::vector<java_annotationt> &get_annotations()
+  {
+    // This cast should be safe as java_annotationt doesn't add data to irept
+    return reinterpret_cast<std::vector<java_annotationt> &>(
+      add(ID_annotations).get_sub());
+  }
+};
+
+class annotated_typet : public typet
+{
+public:
+  std::vector<java_annotationt> &get_annotations()
+  {
+    // This cast should be safe as java_annotationt doesn't add data to irept
+    return reinterpret_cast<std::vector<java_annotationt> &>(
```

Comment:
+1 for using `reinterpret_cast` (and thus something that one can efficiently search for) instead of the C-style cast. Really, we should rid of all of those...

Reply:
This makes me shiver. I'd much prefer a loop/`std::transform` since this bit shouldn't affect performance (in practice it would be iterated over only once). But at least it's not a c-style cast.

Reply:
Just curious: how would that work as really you want to return a reference to the same storage?